tools/integration-tests/tester/tests/testutil.go

```diff
@@ -95,6 +100,271 @@ func CheckForMessageIds(t *testing.T, peers []*framework.Peer, ids map[string]Da
+		if fail {
+			i--
+			continue
+		}
+
+		// attach tx id
+		txIds = append(txIds, txId)
+
+		// let the transaction propagate
+		time.Sleep(3 * time.Second)
+	}
+
+	return
+}
+
+// SendIotaValueMessages sends IOTA token from and to a given peer and returns the transaction ID.
+// The same addresses are used in each round
+func SendIotaValueMessages(t *testing.T, from *framework.Peer, to *framework.Peer, addrBalance map[string]map[balance.Color]int64) (fail bool, txId string) {
+	var sentValue int64 = 100
+	sigScheme := signaturescheme.ED25519(*from.Seed().KeyPair(0))
+	inputAddr := from.Seed().Address(0)
+	outputAddr := to.Seed().Address(0)
+
+	// prepare inputs
+	resp, err := from.GetUnspentOutputs([]string{inputAddr.String()})
+	require.NoErrorf(t, err, "Could not get unspent outputs on %s", from.String())
+
+	// abort if no unspent outputs
+	if len(resp.UnspentOutputs[0].OutputIDs) == 0 {
+		return true, ""
+	}
+	availableValue := resp.UnspentOutputs[0].OutputIDs[0].Balances[0].Value
+
+	//abort if the balance is not enough
+	if availableValue < sentValue {
+		return true, ""
+	}
+
+	out, err := transaction.OutputIDFromBase58(resp.UnspentOutputs[0].OutputIDs[0].ID)
+	require.NoErrorf(t, err, "Invalid unspent outputs ID on %s", from.String())
+	inputs := transaction.NewInputs([]transaction.OutputID{out}...)
+
+	// prepare outputs
+	outmap := map[address.Address][]*balance.Balance{}
+	if inputAddr == outputAddr {
+		sentValue = availableValue
+	}
+
+	// set balances
+	outmap[outputAddr] = []*balance.Balance{balance.New(balance.ColorIOTA, sentValue)}
+	outputs := transaction.NewOutputs(outmap)
+
+	// handle remain address
+	if availableValue > sentValue {
+		outputs.Add(inputAddr, []*balance.Balance{balance.New(balance.ColorIOTA, availableValue-sentValue)})
+	}
+
+	// sign transaction
+	txn := transaction.New(inputs, outputs).Sign(sigScheme)
+
+	// send transaction
+	txId, err = from.SendTransaction(txn.Bytes())
+	require.NoErrorf(t, err, "Could not send transaction on %s", from.String())
+
+	addrBalance[inputAddr.String()][balance.ColorIOTA] -= sentValue
+	addrBalance[outputAddr.String()][balance.ColorIOTA] += sentValue
+
+	return false, txId
+}
+
+// SendColoredValueMessagesOnRandomPeer sends colored token on a random peer and saves the sent token to a map.
+func SendColoredValueMessagesOnRandomPeer(t *testing.T, peers []*framework.Peer, addrBalance map[string]map[balance.Color]int64, numMessages int) (txIds []string) {
+	for i := 0; i < numMessages; i++ {
+		from := rand.Intn(len(peers))
+		to := rand.Intn(len(peers))
+		fail, txId := SendColoredValueMessage(t, peers[from], peers[to], addrBalance)
+		if fail {
+			i--
+			continue
+		}
+
+		// attach tx id
+		txIds = append(txIds, txId)
+
+		// let the transaction propagate
+		time.Sleep(3 * time.Second)
+	}
+
+	return
+}
+
+// SendColoredValueMessage sends a colored tokens from and to a given peer and returns the transaction ID.
```

Comment:
This comment doesn't reflect what the function does, which is:
* Get the first utxo from `from`'s first address
* Accumulate the token amount from all balances on that utxo
* Send [50 uncolored tokens] + [(accumulated token amount - 50) new minted tokens] to `to`'s first address

Describe in the function comment more precisely what this function does.